core/domain/event_services.py

```diff
@@ -92,20 +120,29 @@ def _handle_event(cls, exploration_id, exp_version, aggregated_stats):
 class AnswerSubmissionEventHandler(BaseEventHandler):
     """Event handler for recording answer submissions."""
 
-    EVENT_TYPE = feconf.EVENT_TYPE_ANSWER_SUBMITTED
+    EVENT_TYPE: str = feconf.EVENT_TYPE_ANSWER_SUBMITTED
```

Comment:
```suggestion
    EVENT_TYPE: Final = feconf.EVENT_TYPE_ANSWER_SUBMITTED
```

Reply:
Isn't this `Final`?
Ditto elsewhere `EVENT_TYPE` is used

Reply:
`EVENT_TYPE` is already defined in `BaseEventHandler` and we are just redefining it in all child classes. Also, if we use Final here then MyPy throws an error. Because Final is only used with Constants but here we are redefining it in child classes.

Reply:
No, I have tried!!. Reason:
If we do something like the below:
```python
class BaseEventHandler:
    EVENT_TYPE: str

class StatsEventsHandler(BaseEventHandler):
    EVENT_TYPE: Final = feconf.EVENT_TYPE_ALL_STATS
```

Then in `BaseEventHandler` it is considered as a writable attribute ( non-constant ) but in `StatsEventsHandler` it is considered as readable only attribute ( because of Final ), so here again, we are contradicting the same constant and non-constant idea.